comp/metadata/softwareinventory/impl/inventorysoftware.go

```diff
@@ -192,10 +265,22 @@ func (is *softwareInventory) getPayload() marshaler.JSONMarshaler {
 // This method is used by the HTTP endpoint to serve software inventory data
 // in JSON format for external consumption.
 func (is *softwareInventory) writePayloadAsJSON(w http.ResponseWriter, _ *http.Request) {
-	json, err := is.GetAsJSON()
+	json, err := is.getPayload().MarshalJSON()
 	if err != nil {
 		httputils.SetJSONError(w, err, 500)
 		return
 	}
 	_, _ = w.Write(json)
 }
+
+// FlareProvider returns a flare provider for the software inventory component
+func (is *softwareInventory) FlareProvider() flaretypes.Provider {
+	return flaretypes.NewProvider(
+		func(fb flaretypes.FlareBuilder) error {
+			payload := is.getPayload()
+			if payload == nil {
+				return nil
+			}
```

Comment:
Could we add a file to the flare saying that the product is disabled or with an error message instead ?